### PR TITLE
extract stable-layout crate

### PR DESCRIPTION
**Pull request by kevinheavey**
#### Problem
stable-layout is a blocker for pulling out `solana_program::program`

#### Summary of Changes
Move it to its own crate. Re-export with deprecation notice

### CODE REVIEW

File: sdk/stable-layout/src/lib.rs
Comment (joncinque) [inline sdk/stable-layout/src/lib.rs:1]: This file has a `#![doc(hidden)]` attribute at the top, can we remove that to get docs on the crate? Although it's internal, it'll still be more useful to have docs.rs pages than not

**Review by joncinque — COMMENTED** [review]
Looks great! Just the one comment about `doc(hidden)`

**Review by joncinque — COMMENTED** [review]
Perfect! @yihau can you accept ownership of `solana-stable-layout`?

**Comment by yihau** [comment]
✅

**Review by joncinque — APPROVED** [review]

**Pull request merged**